--- ./../../../setup/src/Magento/Setup/Model/ObjectManagerProvider.php	2023-10-24 10:29:14
+++ ./../../../setup/src/Magento/Setup/Model/ObjectManagerProvider.php	2023-10-24 10:30:40
@@ -60,9 +60,11 @@
             $initParams = $this->serviceLocator->get(InitParamListener::BOOTSTRAP_PARAM);
             $factory = $this->getObjectManagerFactory($initParams);
             $this->objectManager = $factory->create($initParams);
-            if (PHP_SAPI == 'cli') {
-                $this->createCliCommands();
-            }
+// Disabled early CLI command initialisation, to prevent DB access from bin/magento setup:install, which breaks
+// integration tests. This is because some classes being injected in CLI command constructors trigger DB access.
+//            if (PHP_SAPI == 'cli') {
+//                $this->createCliCommands();
+//            }
         }
         return $this->objectManager;
     }